pennylane/interfaces/batch/__init__.py

```diff
@@ -59,15 +59,18 @@ def set_shots(device, shots):
     >>> set_shots(dev, shots=100)(lambda: dev.shots)()
     100
     """
+    if shots == device.shots:
+        yield
+        return

```

Comment:
Now, if `shots == device.shots`, we yield and do nothing!

Reply:
Ah good idea!